granulate_utils/linux/cgroups/cpu_cgroup.py

```diff
@@ -19,7 +19,9 @@ def set_cpu_limit_cores(self, cores: float) -> None:
 
     def get_cpu_limit_cores(self) -> float:
         period = int(self.read_from_control_file(self.cfs_period_us))
-        return int(self.read_from_control_file(self.cfs_quota_us)) / period
+        quota = int(self.read_from_control_file(self.cfs_quota_us))
+        # if quota is set to -1 it means this cgroup is unlimited
+        return quota / period if quota != -1 else -1.0
```

Comment:
It would be nicer to return `None` instead of `-1` but linux likes using `-1` for these tunables
I'm fine with keeping this as `-1` to represent unlimited

Reply:
that makes sense when you are writing in C and using the sign bit to represent a null value, but we are in python here so testing for `< 0` is more confusing than testing `is None`.

Reply:
@d3dave , I'd rather not change the return value to `None` as it changes the API since the function will return `Optional[Float]` instead of `float`. I don't mind changing it but I'd like to talk offline about what else I need to check for this change as I'm not entirely sure.

Reply:
Ok. No real issue with keeping it -1.0. The only two callsites I found don't test the value anyway.